src/rdkafka_admin.c

```diff
@@ -4808,7 +4807,293 @@ void rd_kafka_DescribeAcls(rd_kafka_t *rk,
+                for(itr=0;itr<num_credentials;itr++){
+                        int8_t mechanism;
+                        int32_t iterations;
+                        rd_kafka_buf_read_i8(reply,&mechanism);
+                        rd_kafka_buf_read_i32(reply,&iterations);
+                        rd_kafka_buf_skip_tags(reply);
+                        rd_kafka_ScramCredentialInfo_t *scram_credential = rd_kafka_UserScramCredentialsDescription_get_scramcredentialinfo(description,itr);
+                        rd_kafka_ScramCredentialInfo_set_iteration(scram_credential,iterations);
+                        rd_kafka_ScramCredentialInfo_set_mechanism(scram_credential,mechanism);
+                }
+                rd_list_add(&rko_result->rko_u.admin_result.results,description);
+        }
+        *rko_resultp = rko_result;
+
+        return RD_KAFKA_RESP_ERR_NO_ERROR;
+        
+}
+void rd_kafka_DescribeUserScramCredentials(rd_kafka_t *rk,
+                           char **users,
+                           size_t num_users,
+                           const rd_kafka_AdminOptions_t *options,
+                           rd_kafka_queue_t *rkqu){
 
+        rd_kafka_op_t *rko;
+
+        static const struct rd_kafka_admin_worker_cbs cbs = {
+            rd_kafka_DescribeUserScramCredentialsRequest,
+            rd_kafka_DescribeUserScramCredentialsResponse_parse,
+        };
+
+        rko = rd_kafka_admin_request_op_new(rk, RD_KAFKA_OP_DESCRIBEUSERSCRAMCREDENTIALS,
+                                            RD_KAFKA_EVENT_DESCRIBEUSERSCRAMCREDENTIALS_RESULT,
+                                            &cbs, options, rkqu->rkqu_q);
+
+        rd_list_init(&rko->rko_u.admin_request.args, num_users,
+                     rd_kafkap_str_destroy);
+        int i;
+        for(i =0;i<num_users;i++){
+                rd_list_add(&rko->rko_u.admin_request.args,
+                                rd_kafkap_str_new(users[i],-1));
+        }
+        rd_kafka_q_enq(rk->rk_ops, rko);
+
+}
+rd_kafka_resp_err_t rd_kafka_AlterUserScramCredentialsRequest(rd_kafka_broker_t *rkb,
+    const rd_list_t *user_scram_credential_alterations,
+    rd_kafka_AdminOptions_t *options,
+    char *errstr,
+    size_t errstr_size,
+    rd_kafka_replyq_t replyq,
+    rd_kafka_resp_cb_t *resp_cb,
+    void *opaque){
+        
+        rd_kafka_buf_t *rkbuf;
+        int16_t ApiVersion = 0; 
+        int features;
+        /*Add the correct version of Api*/
+        ApiVersion = rd_kafka_broker_ApiVersion_supported(
+            rkb, RD_KAFKAP_DescribeUserScramCredentials, 0, 0, &features);
+        if (ApiVersion == -1) {
+                rd_snprintf(errstr, errstr_size,
+                            "AlterUserScramCredentials API (KIP-554) not supported "
+                            "by broker");
+                return RD_KAFKA_RESP_ERR__UNSUPPORTED_FEATURE;
+        }
+        int num_alterations = rd_list_cnt(user_scram_credential_alterations);
```

Comment:
nit: move this and other variable declaration stuff up in function

Reply:
sure !